src/control/server/instance_test.go

```diff
@@ -80,6 +80,28 @@ func TestIOServerInstance_NotifyReady(t *testing.T) {
 	waitForIosrvReady(t, instance)
 }
 
+func getTestBioErrorReq(t *testing.T, sockPath string, idx uint32, tgt int32, unmap bool, read bool, write bool) *srvpb.BioErrorReq {
+	return &srvpb.BioErrorReq{
+		DrpcListenerSock: sockPath,
+		InstanceIdx:      idx,
+		TgtId:            tgt,
+		UnmapErr:         unmap,
+		ReadErr:          read,
+		WriteErr:         write,
+	}
+}
+
+func TestIOServerInstance_BioError(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	instance := getTestIOServerInstance(log)
+
+	req := getTestBioErrorReq(t, "/tmp/instance_test.sock", 0, 0, false, false, true)
+
+	instance.BioErrorNotify(req)
+}
```

Comment:
This test doesn't seem to actually test anything as far as I can tell. You could check the buffer for some kind of expected output, maybe?

Reply:
updated test to search for expected I/O error notification